pancancer-arch-3/src/main/java/info/pancancer/arch3/worker/WorkerRunnable.java

```diff
@@ -364,6 +350,34 @@ private WorkflowResult launchJob(String message, Job job, String seqwareEngine,
         return workflowResult;
     }
 
+    private String writeDockerRunnerScript(List<String> dockerArgs) throws IOException {
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("#! /bin/bash\n");
+        sb.append("docker run ");
+        for (String arg : dockerArgs) {
+            sb.append(arg).append(" ");
+        }
+        sb.append(" && cat /home/" + this.userName + "/worker.cid >> /home/" + this.userName + "/successful_container_cids");
```

Comment:
It looks like we're using a Bash script run_worker_daemon.sh to run a Java Program (the worker) in order to run a Bash script here to merely to append container IDs?

Can you do this last layer without Bash?

Also, does the cid portion work when the "docker run" command fails? I would expect that to not work
> mkdir d && cd d 
> ...  
> If the mkdir application was successful (exit code 0), then Bash will execute the next command, cd d. If mkdir d failed, and returned a non-0 exit code, Bash will skip the next command, and we will stay in the current directory

http://mywiki.wooledge.org/BashGuide/TestsAndConditionals#Exit_Status